### DISCUSSION TITLE
Lesson 7: Pushing to github

### DISCUSSION THREAD

**Discussion by codeplay22** [Cyfrin/foundry-full-course-cu#1744] [Q&A] [answered]
i have just pushed my code once and it's shows 3 commit's why?

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/116997721/7c8fa30d-e062-48d3-b5ba-82b0189d7feb)

**Comment by EngrPips** [comment]
I assume the push action and the commit action are not the same. The commit action commits changes to git, while the push action pushes code to GitHub. So, I will say that makes sense.

**Comment by urosognjenovic** [comment, marked as answer]
As much as I know, Foundry automatically creates a commit on initialization of a new project and on adding a new dependency so that's two commits. Even though forge-std is a built in library, I guess it is still counted as a dependency. The third commit is the one you made before pushing to GitHub I guess.

If you remember, Patrick said in the course to always add "--no-commit" when installing a dependency. That's exactly because of the default behavior of Foundry to create a commit after installing a dependency so you have to instruct it not to create a new commit.

**Reply by EngrPips** [reply]
Awesome, Thanks for the detailed explanation. I Learnt a lot.

**Reply by urosognjenovic** [reply]
You're welcome!

**Reply by EngrPips** [reply]
Indeed

**Reply by codeplay22** [reply]
Hi @thebraveman17 
Thanks a lot for you answer! Now i understood why we write ```--no-commit``` with every dependency we install.
So if i write ```forge init --no-commit``` then it will not count as a commit, right?

**Reply by EngrPips** [reply]
Yeah i believe

**Reply by urosognjenovic** [reply]
@codeplay22 That's right, you can use ```--no-commit``` to instruct Foundry not to create a commit.

On the first picture you can see the git log (commits) after creating a new project using ```forge init```. As you can see, there are two commits as I explained above.

On the second picture you can see the output of ```git log``` after creating a new project using ```forge init --no-commit```. As you can see, there are no commits.
<img width="457" alt="Screenshot 2024-05-25 at 08 52 47" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/104977001/1c1c5493-08d5-49a1-a388-01cc9d81a1cb">
<img width="479" alt="Screenshot 2024-05-25 at 08 55 05" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/104977001/ec904fac-83ae-4a2e-8a89-762b9d025e86">

**Reply by codeplay22** [reply]
@thebraveman17 Thanks!
